Tasks/Common/nuget-task-common/NuGetToolRunner.ts

```diff
@@ -155,6 +156,78 @@ export function locateNuGetExe(userNuGetExePath: string): string {
     return toolPath;
 }
 
+function isHosted(): boolean {
+    // not an ideal way to detect hosted, but there isn't a variable for it, and we can't make network calls from here
+    // due to proxy issues.
+    const collectionUri = tl.getVariable("System.TeamFoundationCollectionUri");
+    return /\.visualstudio\.com$/i.test(collectionUri);
```

Comment:
Yeah, I agree. I didn't want to put special knowledge of those internal concerns in the task itself, part of the reasoning for the override variables.